File: frontend/src/api/styleType.ts
import type {  ResponseSuccess, ResponseSuccessWithData, ResponseWitMetaType } from "@/core/models/base";
import { privateHttp } from "./http";
import type { CreateStyleType, StyleType } from "@/core/models/styleType";


export function getStyleTypes(): Promise<ResponseWitMetaType<StyleType[]>> {
    return privateHttp.get("/style-types")
}

export function createNewStyleType(payload: CreateStyleType): Promise<ResponseSuccessWithData<StyleType>> {
    return privateHttp.post("/style-type/create", payload)
}

export function deleteStyleType(style_type_id: number): Promise<ResponseSuccess> {
    return privateHttp.delete(`/style-type/${style_type_id}`)
}

export function getStyleTypeById(style_type_id: number): Promise<ResponseSuccessWithData<StyleType>> {
    return privateHttp.get(`/style-type/${style_type_id}`)
}

export function updateStyleType(payload: StyleType): Promise<ResponseSuccessWithData<StyleType>> {
    return privateHttp.put("/style-type/update", payload)
}

File: backend/internal/routes/styleType.go
package routes

import (
	"chairTime/constant"
	"chairTime/internal/app"
	styleTypeController "chairTime/internal/controllers/styleType"

	"github.com/labstack/echo/v4"
)

func StyleTypeRoute(app *app.Application, groupRoute echo.Group) {
	groupRoute.GET("/style-types", app.RouteHandler(styleTypeController.GetStyleTypes))
	groupRoute.POST("/style-type/create", app.RouteHandler(styleTypeController.CreateStyleType), app.Authorization([]int{constant.AdminRoleId}))
	groupRoute.GET("/style-type/:style_type_id", app.RouteHandler(styleTypeController.GetStyleTypeById), app.Authorization([]int{constant.AdminRoleId}))
	groupRoute.DELETE("/style-type/:style_type_id", app.RouteHandler(styleTypeController.DeleteStyleType), app.Authorization([]int{constant.AdminRoleId}))
	groupRoute.PUT("/style-type/update", app.RouteHandler(styleTypeController.UpdateStyleType), app.Authorization([]int{constant.AdminRoleId}))
}
